src/flake8_bandit/settings.rs

```diff
@@ -0,0 +1,83 @@
+//! Settings for the `flake8-bandit` plugin.
+
+use ruff_macros::ConfigurationOptions;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+fn default_tmp_dirs() -> Vec<String> {
+    ["/tmp", "/var/tmp", "/dev/shm"]
+        .map(std::string::ToString::to_string)
+        .to_vec()
+}
+
+#[derive(
+    Debug, PartialEq, Eq, Serialize, Deserialize, Default, ConfigurationOptions, JsonSchema,
+)]
+#[serde(
+    deny_unknown_fields,
+    rename_all = "kebab-case",
+    rename = "Flake8BanditOptions"
+)]
+pub struct Options {
+    #[option(
+        default = "[\"/tmp\", \"/var/tmp\", \"/dev/shm\"]",
+        value_type = "Vec<String>",
+        example = "hardcoded_tmp_directory = [\"/foo/bar\"]"
+    )]
+    /// List of directories that are considered temporary.
+    pub hardcoded_tmp_directory: Option<Vec<String>>,
+    #[option(
+        default = "[]",
+        value_type = "Vec<String>",
+        example = "extend_hardcoded_tmp_directory = [\"/foo/bar\"]"
+    )]
+    /// List of directories that are considered temporary.
+    /// These directories are added to the list in
+    /// `hardcoded_tmp_directory`.
+    pub hardcoded_tmp_directory_extend: Option<Vec<String>>,
+}
+
+#[derive(Debug, Hash)]
+pub struct Settings {
+    pub hardcoded_tmp_directory: Vec<String>,
```

Comment:
You could consider just having `hardcoded_tmp_directory` (removing `hardcoded_tmp_directory_extend` from `Settings`), and then concatenating the two in `From<Options>`.